### PR TITLE
Remove `pip install` from CI jobs where possible

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

Python dependencies are already installed in `datadog-agent-buildimages` for a bit (see https://github.com/DataDog/datadog-agent-buildimages/pull/197), so `pip install` is essentially a no-op. The PR even mentions "(possibly) Remove the pip install requirements.txt lines from the Gitlab CI", so this is what we're doing here. This should make pipelines *a little bit* faster.

### Describe how to test/QA your changes

CI does not break 😁 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
Is there a reason why we are removing the install from these jobs in particular and not the rest? There are a few places where we still would use it after this PR:

- https://github.com/DataDog/datadog-agent/blob/fabb98b78dd1ccd044d3f4453f6bb5ea53ba56ad/.gitlab/e2e.yml#L14
- container_build jobs
- package_build jobs for dmg and apk
- winbuildscripts
- (probably some more)

It's fine to leave it in those for now, but I would like to understand why the jobs on this PR are particularly interesting

**Comment by mx-psi** [comment]
> The ones left are not using the buildimages where we already have the python deps

Ah, right. We also install on datadog-agent-builders images, so there are probably a few more places where you can remove it, but we can focus this PR on datadog-agent-buildimages only

**Review by mx-psi — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -133,10 +133,10 @@ variables:
...
   RELEASE_VERSION_6: nightly
   RELEASE_VERSION_7: nightly-a7
-  DATADOG_AGENT_BUILDIMAGES: v7341670-fda66f1
+  DATADOG_AGENT_BUILDIMAGES: v7556641-c061240
   DATADOG_AGENT_BUILDERS: v6814105-5ec7b27
   DATADOG_AGENT_WINBUILDIMAGES: v7341670-fda66f1
-  DATADOG_AGENT_ARMBUILDIMAGES: v7341670-fda66f1
+  DATADOG_AGENT_ARMBUILDIMAGES: v7556641-c061240
   DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v7341670-fda66f1
   DATADOG_AGENT_NIKOS_BUILDIMAGES: v7341670-fda66f1
```
Comment (L3n41c) [inline .gitlab-ci.yml:141]: `DATADOG_AGENT_BUILDIMAGES`, `DATADOG_AGENT_WINBUILDIMAGES`, `DATADOG_AGENT_ARMBUILDIMAGES`, `DATADOG_AGENT_SYSPROBE_BUILDIMAGES` and `DATADOG_AGENT_NIKOS_BUILDIMAGES` used to have the same tag value.
I wonder if we should update them all.

**Review by L3n41c — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (mx-psi) [reply .gitlab-ci.yml:141]: Good catch, yes, if possible they should have the same value.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**